src/components/responsive-flyout/ResponsiveFlyout.tsx

```diff
@@ -0,0 +1,48 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+// @ts-ignore flow
+import { Flyout, Overlay, OverlayHeader } from '../flyout';
+// @ts-ignore flow
+import { FlyoutProps } from '../flyout/Flyout';
+
+import './styles/ResponsiveFlyout.scss';
+
+export interface ResponsiveFlyoutProps {
+    /** Components to render in the overlay */
+    children?: React.ReactNode;
+    /** Set className to the overlay wrapper */
+    className?: string;
+    /** Disable responsiveness behavior for higher-level components that want control */
+    enabled?: boolean;
+    /** Custom button to trigger for opening/closing the flyout */
+    flyoutButton: React.ReactElement;
```

Comment:
Should this be `React.ReactNode`?

Reply:
I understood ReactNode could be more than just an object with type and props. ReactNode can be a fragment, array of nodes, string, etc. I don't see flyoutButton or header being more than just an element.